client/components/domains/contact-details-form-fields/index.jsx

```diff
@@ -120,10 +120,13 @@ export class ContactDetailsFormFields extends Component {
 		this.shouldAutoFocusAddressField = false;
 	}
 
+	// `formState` forces multiple updates to `this.state`
+	// This is an attempt limit the redraws to only what we need.
 	shouldComponentUpdate( nextProps, nextState ) {
 		return (
 			! isEqual( nextState.form, this.state.form ) ||
 			! isEqual( nextProps.labelTexts, this.props.labelTexts ) ||
+			! isEqual( nextState.phoneCountryCode, this.state.phoneCountryCode ) ||
```

Comment:
It is in general not a good practice to use deep equal when not necessary. In this case, a simple in-equality expression, `nextState.phoneCountryCode !== this.state.phoneCountryCode` would cost just one inequality comparison, while [lodash isEqual() would cost two extra function calls](https://github.com/lodash/lodash/blob/4.17.10/lodash.js#L11530) before arriving the underlying equality expression.

Although it would make this look less consistent with others, I'd like to propose that we just use a simple inequality expression here :)

Reply:
Agree 💯 % 
That was a lazy copy paste on my behalf. Glad that you discovered it. 
🙇